km61/Servatmand_Mariam/laboratory1.sql

```diff
@@ -1,2 +1,129 @@
 -- LABORATORY WORK 1
 -- BY Servatmand_Mariam
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+робити вибірки з таблиць
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE USER Servatmand INDENTIFIED BY Servatmand
+DEFAULT TABLESPACE USERS
+TEMPORARY TABLESPACE TEMP
+
+ALTER Servatmand QUOTA 100 MB ON USERS
+ GRANT "CONNECT" TO Servatmand
+GRANT SELECT ANY TABLE TO USER
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Громадянин України має власне житло та автомобіль.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
```

Comment:
You must have three entities: Citizen, House_and_car, Citizen_house_car.
Each entity must have at least 4 columns/fields.

Entity Citizen must have fields: citizen_name VARCHAR(30), phone_number VARCHAR(13), person_id NUMBER(12), age NUMBER(3).
Entity House must have fields: street VARCHAR(15), floor_number NUMBER(3), resident_number NUMBER(4), building_year NUMBER(4).
Entity Car must have fields: car_id VARCHAR(20), car_model VARCHAR(15), car_type VARCHAR(15), release_year NUMBER(4).
Entitiy Citizen_house_car must have fields: person_id NUMBER(12), citizen_name VARCHAR(30), street VARCHAR(15), car_id VARCHAR(20).
Primary key for Citizen is 'person_id' and 'citizen_name', for House it's 'street', for Car it's 'car_id'.
Primary key for Citizen_house_car is 'person_id', 'citizen_name' and 'street'.
Foreign keys in the last table should be referenced by primary keys in entities Car, House and Citizen.

Regular expresions for each column in table:

Citizen table:
citizen_name - [A-Z][a-z]{2,14}\s[A-Z][a-z]{2,14}
Examples: Bob Marley, James Hetfield, Corey Taylor.
phone_number - +\d{11,14}
Examples: +14155552671, +13284734341, +19750736593
person_id - \d{10,12}
Examples: 936507643971, 605446309511, 985385736193, 785385630571.
age - [1-9]{1}\d{1,2}
Examples: 18, 74, 72.

House table:
street - [1-9]{1}[0-9]{1,2}\s[A-Z]{1}[a-z]{5,20}\s(St)
Examples: 43 Addams St, 2 Bennington St, 13 Richardson St.
floor_number - [1-9]{1}\d{1}
Examples: 1, 2, 45, 9, 15, 99.
resitent_number: [1-9]{1}\d{1,2}
Examples: 4, 14, 234, 632, 130, 35.
building_year - (19\d{2}|200\d{1}|201[0-8])
Examples: 1993, 2012, 1935.

Car table:
car_id - [A-Z]{2,3}-[0-9]{3}
Examples: GD-023, ANW-831, FE-933.
car_model: [A-Z]{1}[a-z]{5,12}
Examples: Nissan, Toyota, Renault.
car_type: [A-Z]{1}[a-z]{5,12}
Examples: Sports, Mini, Minivan, Regular.
release_year: (19\d{2}|200\d{1}|201[0-8])
Examples: 1943, 1966, 1989, 2012, 2004.